src/data/array/lemmas.lean

```diff
@@ -153,13 +153,13 @@ begin
   { exact ⟨ll.symm ▸ h, to_list_nth_le _ _ _⟩ }
 end
 
-theorem write_to_list {i v} : (a.write i v).to_list = a.to_list.update_nth i.1 v :=
+theorem write_to_list {i v} : (a.write i v).to_list = a.to_list.update_nth i v :=
 list.ext_le (by simp) $ λ j h₁ h₂, begin
   have h₃ : j < n, {simpa using h₁},
   rw [to_list_nth_le _ h₃],
   refine let ⟨_, e⟩ := list.nth_eq_some.1 _ in e.symm,
-  by_cases ij : i.1 = j,
-  { subst j, rw [show fin.mk i.val h₃ = i, from fin.eq_of_veq rfl,
+  by_cases ij : (i : ℕ) = j,
+  { subst j, rw [show (⟨(i : ℕ), h₃⟩ : fin _) = i, from fin.eq_of_veq rfl,
```

Comment:
Doesn't `fin.mk` still exist? This is a bit ugly.

Reply:
`fin.mk` exists. Maybe it should be made reducible? Right now this line was broken.